daemon/cluster/cluster.go

```diff
@@ -1030,14 +1061,91 @@ func (c *Cluster) managerStats() (current bool, reachable int, unreachable int,
 	return
 }
 
+func validateInitRequest(req *types.InitRequest) error {
```

Comment:
nit: This does mire than just validate thr request. It may also change it.

`validateAndSanitizeInitRequest`?